Realm/Realm/Realm.cs

```diff
@@ -595,7 +595,81 @@ public void Add<T>(IEnumerable<T> objs, bool update = false)
 
             foreach (var obj in objs)
             {
-                AddInternal(obj, typeof(T), update);
+                AddInternal(obj, obj.GetType(), update);
+            }
+        }
+
+        /// <summary>
+        /// This <see cref="Realm"/> will start managing a <see cref="AsymmetricObject"/> which has been created as a standalone object.
+        /// </summary>
+        /// <typeparam name="T">
+        /// The Type T must not only be a <see cref="AsymmetricObject"/> but also have been processed by the Fody weaver,
+        /// so it has persistent properties.
+        /// </typeparam>
+        /// <param name="obj">Must be a standalone <see cref="AsymmetricObject"/>, <c>null</c> not allowed.</param>
+        /// <exception cref="RealmInvalidTransactionException">
+        /// If you invoke this when there is no write <see cref="Transaction"/> active on the <see cref="Realm"/>.
+        /// </exception>
+        /// <exception cref="RealmObjectManagedByAnotherRealmException">
+        /// You can't manage an object with more than one <see cref="Realm"/>.
+        /// </exception>
+        /// <exception cref="ObjectDisposedException">
+        /// If you invoke this method on a closed <see cref="Realm"/>.
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        /// If you pass a null argument to this method.
+        /// </exception>
+        /// <remarks>
+        /// If the object is already managed by this <see cref="Realm"/>, this method does nothing.
+        /// This method modifies the object in-place, meaning that after it has run, <see cref="AsymmetricObject"/> will be managed.
+        /// Once an <see cref="AsymmetricObject"/> becomes managed dereferencing the original <see cref="AsymmetricObject"/>
```

Comment:
I don't understand this part.

Reply:
Could you be more specific?

Reply:
"dereferencing the original reference throws an exception" - I don't believe this is the case, why would it throw?

Reply:
Once the `AsymmetricObject` becomes managed, you get a runtime error if trying to use that reference.
We all agreed that this is how it should be. Why do you disagree?
The test for this is `AccessAsymmetricObjAfterAddedToRealm_Throws`

Reply:
Conversation taken offline. Resolved.